### ISSUE TITLE
[bug] generated settings.yml missing Macos.version "14.2", causes build failures

### ISSUE THREAD

**Issue by garyo** [conan-io/conan#15854]
### Environment details

* M1 Macbook Pro
* MacOS Sonoma 14.2.1
* conan 2.1.0
* python 3.11.5

### Steps to reproduce

Run conan initially. It will generate a `~/.conan2/settings.yml` with the latest version of Macos being "14.1", but the current version is actually 14.2. But it will put "os.version=14.2" into the default profile when you run "conan install". This causes conan to fail with "ERROR: Invalid setting '14.2' is not a valid 'settings.os.version' value."

Workaround is to manually add "14.2" to the OS versions listing in `settings.yml`.

**Comment by jcar87** [comment]
Hi @garyo - thanks for providing more details.

I just checked `cmake-conan` and it should only add `os.version` to the profile if you have `CMAKE_OSX_DEPLOYMENT_TARGET` defined, is this the case for your project?

Otherwise it should be left undefined for Macos

**Comment by garyo** [comment]
I do, but it is defined to "11.0":
```cmake
if(APPLE)
  set(CMAKE_OSX_DEPLOYMENT_TARGET
      "11.0"
      CACHE STRING "MacOS Deployment Target")
  ...
```
I suspect (?) `cmake-conan` runs before that is defined. 

If you'd like a simple repro case, I'm using a template project I created a while ago. Check out https://github.com/garyo/cmake-conan-plugin-template and just run `./build.sh` in there. It will show the error. That project does not set `CMAKE_OSX_DEPLOYMENT_TARGET`.

**Comment by memsharded** [comment]
>  I believe that's the recommended cmake/conan integration. 

I'd like to clarify about this.
This is recommended if you have the need that your developers work only calling ``cmake``, because for some reason they cannot call ``conan`` (like strong dependency on just working from CLion IDE, without even touching a terminal).

Otherwise, the "canonical" Conan + CMake process would be more like:
```
$ conan install ... 
$ cmake --preset conan-default
```

Separating the conan installation of dependencies from the actual cmake build. This is better in many regards, more control, more visibility, developers taking ownership of failures and identifying fixes and improvements instead of just "the build is broken" complaining to someone else...

**Comment by garyo** [comment]
> > I believe that's the recommended cmake/conan integration.
> 
> I'd like to clarify about this. This is recommended if you have the need that your developers work only calling `cmake`, because for some reason they cannot call `conan` (like strong dependency on just working from CLion IDE, without even touching a terminal).
> 
> Otherwise, the "canonical" Conan + CMake process would be more like:
> 
> ```
> $ conan install ... 
> $ cmake --preset conan-default
> ```
> 
> Separating the conan installation of dependencies from the actual cmake build. This is better in many regards, more control, more visibility, developers taking ownership of failures and identifying fixes and improvements instead of just "the build is broken" complaining to someone else...

Yeah, I get that. I figured it's just one more hurdle for someone using/building the project to have to learn & understand, when they're more-or-less used to calling cmake. Especially since the conan cmd line invocation is usually fairly long and specific; you pretty much _have_ to wrap it up into a shell script as I did in that sample project, unless you can get cmake to call it.

In the sample project I linked above, I did provide a shell script that invokes conan manually first (see build-conan-ninja.sh), and that does not suffer from the bug I'm reporting here -- it does not create a file `conan_host_profile` and the build completes normally. To be honest, I don't fully understand all the interactions here between conan, cmake and the integration file, so if you folks say there is nothing to be fixed, I'll just keep my `settings_user.yml`.

**Comment by memsharded** [comment]
>  I figured it's just one more hurdle for someone using/building the project to have to learn & understand, when they're more-or-less used to calling cmake.

That is the doubled edged sword. If they are using a tool like Conan they have to learn and understand that tool, in the same way the have to understand git, understand the IDE, understand linters or sanitizers errors, etc. Otherwise, trying to just isolate devs from a tool means they will get blocked when some small detail is not working, and they will just rely on someone else, maybe you, the typical "build-engineer" or "devops manager" with the "build is broken, please fix it", creating bottlenecks. So it is kind of appealing in the short term, but this is not great in mid term.

FYI, the 14.2 is going to be added to the default settings in https://github.com/conan-io/conan/pull/15859, for next 2.2 release.

Still, it would be great to understand why ``cmake-conan`` integration is adding that, because it is not expected that adding ``os.version`` is the default, only when users opt-in via the CMAKE_OSX_DEPLOYMENT_TARGET, so that might be a but in ``cmake-conan`` we would like to check.

**Comment by jcar87** [comment]
Have added the missing versions to `settings.yml`, however I'm reopening the issue and transferring to `cmake-conan` to better investigate the behaviour
